src/main/java/org/traccar/database/OpenIdProvider.java

```diff
@@ -82,15 +91,47 @@ public OpenIdProvider(Config config, LoginService loginService) {
 
         try {
             callbackUrl = new URI(config.getString(Keys.WEB_URL, "") + "/api/session/openid/callback");
-            authUrl = new URI(config.getString(Keys.OPENID_AUTHURL, ""));
-            tokenUrl = new URI(config.getString(Keys.OPENID_TOKENURL, ""));
-            userInfoUrl = new URI(config.getString(Keys.OPENID_USERINFOURL, ""));
             baseUrl = new URI(config.getString(Keys.WEB_URL, ""));
+
+            if (
```

Comment:
I would simplify this to just take `OPENID_ISSUERURL` if it's set. Otherwise assume that separate URLs are provided.